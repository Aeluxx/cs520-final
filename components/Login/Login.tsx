// Copied from
// https://github.com/mui-org/material-ui/blob/next/docs/src/pages/getting-started/templates/sign-in/SignIn.tsx

import Button from '@mui/material/Button'
import CssBaseline from '@mui/material/CssBaseline'
import TextField from '@mui/material/TextField'
import FormControlLabel from '@mui/material/FormControlLabel'
import Checkbox from '@mui/material/Checkbox'
import Link from '@mui/material/Link'
import Grid from '@mui/material/Grid'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import Container from '@mui/material/Container'
import { useState } from 'react'
import useAuth from '../../hooks/useAuth'
import { FormEvent } from 'react'
import { useRouter } from 'next/router'

// https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
function validateEmail(email: any) {
  const re =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return re.test(String(email).toLowerCase())
}

const Login = () => {
  const [errMessage, setErrMessage] = useState<string>('')
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const router = useRouter()
  const { login } = useAuth()

  const handleLogin = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    login({ email, password })
      .then(() => router.push("/"))
      .catch(() => {
        console.log("Invalid Login");
        setErrMessage("Invalid Login");
      });
  }

  return (
    <Container component='main' maxWidth='xs'>
      <CssBaseline />
      <Box
        sx={{
          marginTop: 6,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}>
        <Typography component='h1' variant='h5'>
          Sign in
        </Typography>
        <Box component='form' onSubmit={handleLogin} noValidate sx={{ mt: 1 }}>
          <TextField
            margin='normal'
            value={email}
            onChange={e => setEmail(e.target.value)}
            required
            fullWidth
            id='email'
            label='Email Address'
            name='email'
            autoComplete='email'
            autoFocus
          />
          <TextField
            margin='normal'
            required
            fullWidth
            value={password}
            onChange={e => setPassword(e.target.value)}
            name='password'
            label='Password'
            type='password'
            id='password'
            autoComplete='current-password'
          />
          <FormControlLabel control={<Checkbox value='remember' color='primary' />} label='Remember me' name='rememberMe' id='rememberMe' />
          <Button type='submit' fullWidth variant='contained' sx={{ mt: 3, mb: 2 }}>
            Sign In
          </Button>
          <Grid container>
            <Grid item xs={12}>
              <Typography variant='body2'>
                Don&apos;t have an account? <Link href='/register'>Create an account</Link>
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <Link href='#' variant='body2'>
                Forgot password?
              </Link>
            </Grid>
          </Grid>
        </Box>
      </Box>
      <Typography variant='caption' color='red'>
        {errMessage}
      </Typography>
    </Container>
  )
}

export default Login
