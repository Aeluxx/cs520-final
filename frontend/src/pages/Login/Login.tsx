// Copied from
// https://github.com/mui-org/material-ui/blob/next/docs/src/pages/getting-started/templates/sign-in/SignIn.tsx

import Button from '@mui/material/Button'
import CssBaseline from '@mui/material/CssBaseline'
import TextField from '@mui/material/TextField'
import FormControlLabel from '@mui/material/FormControlLabel'
import Checkbox from '@mui/material/Checkbox'
import Link from '@mui/material/Link'
import Grid from '@mui/material/Grid'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import Container from '@mui/material/Container'
import { createTheme, ThemeProvider } from '@mui/material/styles'
import { useState } from 'react'
import { useNavigate } from 'react-router'

// function Copyright(props: any) {
//   return (
//     <Typography variant="body2" color="text.secondary" align="center" {...props}>
//       {'Copyright Â© '}
//       <Link color="inherit" href="https://material-ui.com/">
//         Your Website
//       </Link>{' '}
//       {new Date().getFullYear()}
//       {'.'}
//     </Typography>
//   );
// }

// https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
function validateEmail(email: any) {
  const re =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return re.test(String(email).toLowerCase())
}

const Login = props => {
  const [errMessage, setErrMessage] = useState<string>('')

  const navigate = useNavigate()

  const handleSubmit = () => {
    // TODO
  }

  return (
      <Container component='main' maxWidth='xs'>
        <CssBaseline />
        <Box
          sx={{
            marginTop: 6,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}>
          <Typography component='h1' variant='h5'>
            Sign in
          </Typography>
          <Box component='form' onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
            <TextField margin='normal' required fullWidth id='email' label='Email Address' name='email' autoComplete='email' autoFocus />
            <TextField
              margin='normal'
              required
              fullWidth
              name='password'
              label='Password'
              type='password'
              id='password'
              autoComplete='current-password'
            />
            <FormControlLabel control={<Checkbox value='remember' color='primary' />} label='Remember me' name='rememberMe' id='rememberMe' />
            <Button type='submit' fullWidth variant='contained' sx={{ mt: 3, mb: 2 }}>
              Sign In
            </Button>
            <Grid container>
              <Grid item xs>
                <Link href='#' variant='body2'>
                  Forgot password?
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Typography variant='caption' color='red'>
          {errMessage}
        </Typography>
      </Container>
  )
}

export default Login
