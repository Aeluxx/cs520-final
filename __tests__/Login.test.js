import React from 'react';
import '@testing-library/jest-dom';
import Login from '../components/Login/Login';
import ShallowRenderer from 'react-test-renderer/shallow';

let realUseContext;
let useContextMock;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});

afterEach(() => {
    React.useContext = realUseContext;
});

it("Dashboard renders without error", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const login = shallow.render(<Login />);
});

it("Dashboard matches Snapshot", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const login = shallow.render(<Login />);
    const result = shallow.getRenderOutput();
    expect(result).toMatchSnapshot();
});