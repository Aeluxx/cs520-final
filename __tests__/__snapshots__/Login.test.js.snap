// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches Snapshot 1`] = `
<ForwardRef(Container)
  component="main"
  maxWidth="xs"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Login Page
    </title>
    <html
      lang="en"
    />
  </HelmetWrapper>
  <CssBaseline />
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 6,
      }
    }
  >
    <ForwardRef(Typography)
      component="h1"
      variant="h5"
    >
      Sign in
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      component="form"
      noValidate={true}
      onSubmit={[Function]}
      sx={
        Object {
          "mt": 1,
        }
      }
    >
      <ForwardRef(TextField)
        autoComplete="email"
        autoFocus={true}
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        onChange={[Function]}
        required={true}
        value=""
      />
      <ForwardRef(TextField)
        autoComplete="current-password"
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <ForwardRef(FormControlLabel)
        control={
          <ForwardRef(Checkbox)
            color="primary"
            value="remember"
          />
        }
        id="rememberMe"
        label="Remember me"
        name="rememberMe"
      />
      <ForwardRef(Button)
        fullWidth={true}
        sx={
          Object {
            "mb": 2,
            "mt": 3,
          }
        }
        type="submit"
        variant="contained"
      >
        Sign In
      </ForwardRef(Button)>
      <ForwardRef(Grid)
        container={true}
      >
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <ForwardRef(Typography)
            variant="body2"
          >
            Don't have an account? 
            <ForwardRef(Link)
              href="/register"
            >
              Create an account
            </ForwardRef(Link)>
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <ForwardRef(Link)
            href="#"
            variant="body2"
          >
            Forgot password?
          </ForwardRef(Link)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Typography)
    color="red"
    variant="caption"
  >
    
  </ForwardRef(Typography)>
</ForwardRef(Container)>
`;
